### ISSUE TITLE
need a way to call toggle() from slotted footer

### ISSUE THREAD

**Issue by mbokil** [wxsms/uiv#310]
There is one gap in the modal dialog API that I ran into. I wanted a custom footer and used the slot to override the footer. I also wanted to call the modal toggle() function from my custom buttons. I couldn't find a clean way to a do this. Wanted to see if there is a better way to do this. Posting my code here for comment.

Created a wrapper component for Modal:
```
<template>
  <modal v-model="isDialogOpen" ref="modal" @show="init" :size="size" @hide="handleClose" :beforeClose="handleBeforeClose" :dismiss-btn="false" :backdrop="backdrop">
```
In my slotted footer I then use the ref for modal to call toggle:
```
<template v-slot:footer>
      <btn class="btn btn-md btn-success" @click="$refs.modal.toggle(false,'cancel')">Cancel</btn>
      <btn class="btn btn-md btn-success" @click="$refs.modal.toggle(false,'reset')">Reset</btn>
      <btn class="btn btn-md btn-success" @click="$refs.modal.toggle(false,'save')">Save</btn>
</template>
```

**Comment by wxsms** [comment]
You can simply use `@click="isDialogOpen = false"`.  Create your own function in your component if you need to run `beforeClose` before that. That should be a cleaner way.

**Comment by mbokil** [comment]
Thanks for getting back to me. That is a good solution. I ended up using a component wrapper for Modal so I can keep modal code in a separate component to keep the parent component slim. I wanted to take advantage of the msg param so used this solution. Note that I also developed a directive v-modal-moveble so users can drag Modal around. 
`  <access-details-modal v-modal-moveable :backdrop="false" size="lg" :callback="updateScreen" :value.sync="isDialogOpen"></access-details-modal>`

```
handleBeforeClose(msg) {
      if (msg === 'cancel') return true;

      if (msg === 'save') {
        //save logic
      } 

      if (msg === 'reset') {
        //reset logic
      } 
    }
```
